lib/dry/system/plugins/zeitwerk.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+require "dry/system/plugins/zeitwerk/compat_inflector"
+
+module Dry
+  module System
+    module Plugins
+      class Zeitwerk < Module
+        attr_reader :options
+
+        # @api private
+        def initialize(options)
+          @options = options
+          super()
+        end
+
+        # @api private
+        def extended(system)
+          system.before(:configure, &method(:configure_autoloading))
+          system.after(:configure, &method(:setup_autoloader))
+
+          super
+        end
+
+        private
+
+        # Configure autoloading on the container
+        #
+        # @return [self]
+        #
+        # @api private
+        def configure_autoloading(system)
+          require "dry/system/loader/autoloading"
+
+          system.setting :autoloader, reader: true
```

Comment:
I know you're following the pattern of the other plugins here, but now that I look at this, I feel like it'd make more sense to declare the setting as early as possible, i.e. in `#extended` rather than waiting for the before_configure hook. But for the sake of this PR, let's leave this consistent with the other plugins.

So: definitely not a blocker for this PR, but wanted to mention it given it became apparent to me over the course of this review.

@solnic — given you authored most of the existing plugins, is there any reason they declare their settings in before configure hooks, rather than in the standard module extended hook?

Reply:
@timriley it was to ensure that plugin settings are available before configuration takes place, there were cases where doing it in the moment a plugin is required was too early or too late.

Reply:
We're now declaring the settings directly on the container class in `extended`, like the rest of the plugins do after the changes in #207.